napari/components/dims.py

```diff
@@ -140,54 +150,175 @@ def _check_dims(cls, values):
             )
 
         # Check the axis labels tuple has same number of elements as ndim
-        if len(values['axis_labels']) < ndim:
+        labels_ndim = len(values['axis_labels'])
+        if labels_ndim < ndim:
             # Append new "default" labels to existing ones
-            if values['axis_labels'] == tuple(
-                map(str, range(len(values['axis_labels'])))
-            ):
+            if values['axis_labels'] == tuple(map(str, range(labels_ndim))):
                 values['axis_labels'] = tuple(map(str, range(ndim)))
             else:
                 values['axis_labels'] = (
-                    tuple(map(str, range(ndim - len(values['axis_labels']))))
+                    tuple(map(str, range(ndim - labels_ndim)))
                     + values['axis_labels']
                 )
-        elif len(values['axis_labels']) > ndim:
+        elif labels_ndim > ndim:
             values['axis_labels'] = values['axis_labels'][-ndim:]
 
         return values
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # to be removed if/when deprecating current_step
+        self.events.span.connect(self.events.current_step)
+
+    @property
+    def _span_step(self) -> Tuple[float, ...]:
+        return tuple(
+            (
+                # "or 1" ensures degenerate dimension works
+                int(round((low - min_val) / (step or 1))),
+                int(round((high - min_val) / (step or 1))),
+            )
+            for (low, high), (min_val, _), step in zip(
+                self.span, self.range, self.step
+            )
+        )
+
+    @_span_step.setter
+    def _span_step(self, value):
+        self.span = [
+            (
+                min_val + low * step,
+                min_val + high * step,
+            )
+            for (low, high), (min_val, _), step in zip(
+                value, self.range, self.step
+            )
+        ]
+
     @property
-    def nsteps(self) -> Tuple[int, ...]:
-        """Tuple of int: Number of slider steps for each dimension."""
+    def nsteps(self) -> Tuple[float]:
```

Comment:
To me `Tuple[float]` indicates a 1-tuple containing one float (which isn't very useful) whereas `Tuple[float, ...]` indicates a tuple of floats of unknown/varying length.

```suggestion
    def nsteps(self) -> Tuple[float, ...]:
```

There are a few of these changes in this PR, so if we change this one, we should change them all.

Reply:
@andy-sweet is that to you or to mypy though? Cos I kinda disagree here, `Tuple[float]` maps in my brain to e.g. `vector<float>` in C++ or other similar constructs. I don't have much of a stake here, but I sorta like `Tuple[float]` as more concise. I'm pretty rubbish with typing though, so whatever the broader typing community is doing is 👍 from me!

Reply:
I think it's mypy, I just forgot to do that.